### PR TITLE
Remove ARN field from AWSResourceReference

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
We deprecated ARN field from AWSResourceReference in this [PR](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3257) and decided to remove it once v1beta2 API changes are in. This PR removes that field.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes Part of #2355 

**Special notes for your reviewer**:

**Checklist**:
- [x] squashed commits
- [ ] includes documentation
- [X] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: controlplane/eks/api/v1beta1/conversion_test.go [hunk 1]
```diff
@@ -17,15 +17,30 @@ limitations under the License.
...
 
+func fuzzFuncs(_ runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		AWSManagedControlPlaneFuzzer,
+	}
+}
+
+func AWSManagedControlPlaneFuzzer(obj *AWSManagedControlPlane, c fuzz.Continue) {
+	c.FuzzNoCustom(obj)
+	obj.Spec.DisableVPCCNI = false
```
Comment (Ankitasw) [inline controlplane/eks/api/v1beta1/conversion_test.go:40]: A bit doubtful about this change.

**Comment by Ankitasw** [comment]
External vpc test failing, not able to run that test locally 😕

**Comment by Ankitasw** [comment]
Would split into small PRs so that things which are passing can merge soon

File: controlplane/eks/api/v1beta1/conversion_test.go (hunk 1 above)
Comment (Ankitasw) [reply controlplane/eks/api/v1beta1/conversion_test.go]: Removed file from this PR.

**Review by AverageMarcus — COMMENTED** [review]
/lgtm

**Comment by richardcase** [comment]
Non-eks e2e is passing. Lets check the eks side:

/test pull-cluster-api-provider-aws-e2e-eks

**Comment by richardcase** [comment]
We can ignore the API diff error in this instances as we are deliberatrly breaking the contract (but we have conversions):

/override pull-cluster-api-provider-aws-apidiff-main

**Pull request merged**